engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+	dec [hl]
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld a, 2
+	call DrawIntroCharizardGraphic
+	ld a, 64
+	ld [wIntroFrameCounter1], a
+	xor a
+	ld [wIntroFrameCounter2], a
+; fallthrough
+
+IntroScene15:
+; Charizard mouth wide open / fireball starts
+	call Intro_AnimateFireball
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+	dec [hl]
+	ret
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	xor a
+	ld [wIntroFrameCounter1], a
+	ret
+
+IntroScene16:
+; continue fireball / fade out palettes
+	call Intro_AnimateFireball
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	inc [hl]
+	swap a
+	and $07
+	ld e, a
+	ld d, $00
+	ld hl, .palettes
+	add hl, de
+	ld a, [hl]
+	cp $ff
+	jr z, .next
+	ldh [rBGP], a
+	ldh [rOBP0], a
+	ret
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ret
+.palettes:
+	db %11100100
+	db %10010000
+	db %01000000
+	db %00000000
+	db -1
+
+Intro_BlankTilemapAndBGMap:
+	hlcoord 0, 0
+	ld bc, SCREEN_WIDTH * SCREEN_HEIGHT
+.blank_tilemap
+	ld [hl], 0
+	inc hl
+	dec bc
+	ld a, b
+	or c
+	jr nz, .blank_tilemap
+
+	ld hl, wOverworldMapBlocks	; $c600, buffer
+	ld bc, BG_MAP_WIDTH * BG_MAP_HEIGHT
+.blank_bgmap
+	ld [hl], 0
+	inc hl
+	dec bc
+	ld a, b
+	or c
+	jr nz, .blank_bgmap
+
+	ld hl, vBGMap0
+	ld de, wOverworldMapBlocks	; $c600, buffer
+	lb bc, BANK(@), $40
+	call Request2bpp
+	ret
+
+
+Intro_CheckSCYEvent:
+	ldh a, [hSCY]
+	ld c, a
+	ld hl, .Cuelist
+
+.loop
+	ld a, [hli]
+	cp -1
+	ret z
+	cp c
+	jr z, .value_found
+	inc hl
+	inc hl
+	jr .loop
+
+.value_found
+	ld a, [hli]
+	ld h, [hl]
+	ld l, a
+	jp hl
+
+.Cuelist:
+	dbw $87, Intro_BlastoiseAppears
+	dbw $88, Intro_FlashMonPalette
+	dbw $98, Intro_FlashSilhouette
+	dbw $99, Intro_LoadVenusaurPalette
+	dbw $bf, Intro_VenusaurAppears
+	dbw $c0, Intro_FlashMonPalette
+	dbw $d0, Intro_FlashSilhouette
+	dbw $d1, Intro_LoadCharizardPalette
+	db -1
+
+Intro_BlastoiseAppears:
+	call Intro_LoadBlastoiseObject
+	ret
+
+Intro_VenusaurAppears:
+	call Intro_LoadVenusaurObject
+	ret
+
+Intro_FlashMonPalette:
+	ld a, %11100100
+	ldh [rOBP0], a
+	xor a
+	ldh [rBGP], a
+	ret
+
+Intro_FlashSilhouette:
+	ld a, %11111111
+	ldh [rOBP0], a
+	ld a, %00111111
+	ldh [rBGP], a
+	ret
+
+Intro_LoadVenusaurPalette:
+	callab Function9639
+	ret
+
+Intro_LoadCharizardPalette:
+	callab Function963f
+	ret
+
+
+DrawIntroCharizardGraphic:
+	push af
+	hlcoord 0, 6
+	ld c, SCREEN_WIDTH * 8
+	xor a
+.loop1
+	ld [hli], a
+	dec c
+	jr nz, .loop1
+
+	pop af
+	ld e, a
+	ld d, $00
+	ld hl, .charizard_data
+rept 5
+	add hl, de
+endr
+	ld e, [hl]
+	inc hl
+	ld c, [hl]
+	inc hl
+	ld b, [hl]
+	inc hl
+	ld a, [hli]
+	ld h, [hl]
+	ld l, a
+	ld a, e
+
+.loop2_outer
+	push bc
+	push hl
+.loop2_inner
+	ld [hli], a
+	inc a
+	dec c
+	jr nz, .loop2_inner
+
+	pop hl
+	ld bc, SCREEN_WIDTH
+	add hl, bc
+	pop bc
+	dec b
+	jr nz, .loop2_outer
+
+	ld a, 1
+	ldh [hBGMapMode], a
+	call DelayFrame
+	call DelayFrame
+	call DelayFrame
+	xor a
+	ldh [hBGMapMode], a
+	ret
+
+.charizard_data
+	db $00, $08, $08, $22, $c3
```

Comment:
`.loop2_outer` above shows that this data is "first tile ID, width, height, destination" where "destination" is a RAM address (so this would be `db $00, 8, 8 / dw wc322`)